src/agent.spec.ts

```diff
@@ -1,74 +1,173 @@
 import {
-  FindingType,
-  FindingSeverity,
   Finding,
-  HandleTransaction,
+  FindingSeverity,
+  FindingType,
   createTransactionEvent,
-  ethers,
+  HandleTransaction,
+  TransactionEvent,
+  
 } from "forta-agent";
-import agent, {
-  ERC20_TRANSFER_EVENT,
-  TETHER_ADDRESS,
-  TETHER_DECIMALS,
-} from "./agent";
+import { TestTransactionEvent } from "forta-agent-tools/lib/test";
+import provideHandleTransaction from "./agent";
+import { Interface } from "@ethersproject/abi";
+import { FORTA_BOT_REGISTRY,BOT_DEPLOYER_ADDRESS, NEW_AGENT_FUNCTION_SIGNATURE} from "./constants";
+import { BigNumber } from "ethers";
 
-describe("high tether transfer agent", () => {
-  let handleTransaction: HandleTransaction;
-  const mockTxEvent = createTransactionEvent({} as any);
+//mock txs data:
 
+const mockTxEventOne = {
+  agentId:"123456",
+  owner: BOT_DEPLOYER_ADDRESS,
+  chainIds:["56"],
+  metadata: "Dummy info 1",
+}
+
+const mockTxEventTwo = {
+  agentId:"654321",
+  owner: BOT_DEPLOYER_ADDRESS,
+  chainIds:["56"],
+  metadata: "Dummy info 2",
+}
+
+
+describe("bot deployment agent", () => {
+  let handleTransaction: HandleTransaction;
+  let mockTxEvent = createTransactionEvent({} as any);
+  let findings =
   beforeAll(() => {
-    handleTransaction = agent.handleTransaction;
+    handleTransaction = provideHandleTransaction(FORTA_BOT_REGISTRY,BOT_DEPLOYER_ADDRESS,NEW_AGENT_FUNCTION_SIGNATURE);
   });
 
-  describe("handleTransaction", () => {
-    it("returns empty findings if there are no Tether transfers", async () => {
-      mockTxEvent.filterLog = jest.fn().mockReturnValue([]);
-
-      const findings = await handleTransaction(mockTxEvent);
+  const proxyInterface = new Interface([NEW_AGENT_FUNCTION_SIGNATURE]);
 
+  describe("handleTransaction", () => {
+    //no bot deployments case
```

Comment:
The unit test title (the string inside `it(...)`) should be descriptive enough to make comments like this unnecessary.